processing/src/main/java/io/druid/query/topn/DimExtractionTopNAlgorithm.java

```diff
@@ -108,19 +109,42 @@ protected void updateResults(
       TopNResultBuilder resultBuilder
   )
   {
-    for (Map.Entry<Comparable, Aggregator[]> entry : aggregatesStore.entrySet()) {
-      Aggregator[] aggs = entry.getValue();
-      if (aggs != null && aggs.length > 0) {
-        Object[] vals = new Object[aggs.length];
-        for (int i = 0; i < aggs.length; i++) {
-          vals[i] = aggs[i].get();
-        }
+    if (needsResultTypeConversion(params)) {
```

Comment:
Changed this as a part of parameterizing the aggregatesStore, now this is handled within type specific implementations by checking for `valueTransformer == null`